firestore/src/test/java/com/example/firestore/snippets/QueryDataSnippetsIT.java

```diff
@@ -197,6 +197,20 @@ public void testMultipleCursorConditions() throws Exception {
     assertTrue(Objects.equals(result, expectedResults));
   }
 
+  @Test
+  public void testCreateStartAtSnapshotQueryCursor() throws Exception {
+    Query q = queryDataSnippets.createStartAtSnapshotQueryCursor();
+    List<String> results = getResults(q);
+    List<String> expectedResults = Arrays.asList("SF", "LA", "TOK", "BJ");
+    assertEquals(expectedResults, results);
+  }
+
+  @Test
+  public void testPaginateCursor() throws Exception {
+    // Snippet executes it's own query. Failures result in thrown Exceptions
+    queryDataSnippets.paginateCursor();
```

Comment:
This test doesn't verify that the API is returning what's expected, just that it doesn't through any exceptions.

This is my concern with methods that don't return anything, except this one doesn't even log anything to stdout either, so there's no way to catch any bugs or changes from one version to another.

Maybe that's not technically a goal or responsibility with our samples, but if a bug gets through the API and doesn't get caught in our samples, that's going to mean people taking our code and using it in their own projects, and it not working for them either.

Reply:
That's a fair concern, and one I agree with. The reason I went with this over a more involved test was because: 

1. I followed the [golang](https://github.com/GoogleCloudPlatform/golang-samples/blob/3eebbde2743566722299ac5747aa857008f0f9ff/firestore/firestore_snippets/query.go#L203-L220) example
 2. there are only 4 results in the first page, so the second page is technically empty.

Do you have a better way in mind to test it? I could check for an empty list - would that be preferred? I decided against it at the time, mostly because there are similar issues with some snippets in this folder and figured it would be best just to wait until they are modified to fit the new sample format.

Reply:
> figured it would be best just to wait until they are modified to fit the new sample format.

That might be the answer for now, depending on how soon all these samples are going to be updated. Are there any queries that are long enough to be paginated? Or can we just reduce the page element count to force the pagination even if it's not practical to get a page of 2 results?

In any case, I'd like it to at least write its findings to stdout, if not return them to be verified by the test.

Reply:
I've updated the snippet slightly to return the queries and rerun for the tests.